tests/integration/cairo1/hint_locals/start_roll/start_roll_test.cairo

```diff
@@ -0,0 +1,117 @@
+use array::ArrayTrait;
+use result::ResultTrait;
+
+
+#[test]
+fn test_start_roll() {
+    let deployed_contract_address = deploy_contract('simple', ArrayTrait::new()).unwrap();
+    assert(deployed_contract_address != 0, 'deployed_contract_address != 0');
+
+    let result = call(deployed_contract_address, 'check_block_number', ArrayTrait::new()).unwrap();
+    assert(*result.at(0_u32) == -1, *result.at(0_u32));
+
+    start_roll(100, deployed_contract_address);
+    let result = call(deployed_contract_address, 'check_block_number', ArrayTrait::new()).unwrap();
+    assert(*result.at(0_u32) == 100, *result.at(0_u32));
+}
+
+#[test]
+fn test_start_roll_behind_a_proxy() {
+    let target = deploy_contract('simple', ArrayTrait::new()).unwrap();
+    assert(target != 0, 'target != 0');
+
+    let proxy = deploy_contract('proxy', ArrayTrait::new()).unwrap();
+    assert(proxy != 0, 'proxy != 0');
+
+    start_roll(100, target);
+
+    let mut calldata = ArrayTrait::new();
+    calldata.append(target);
+
+    let result = call(proxy, 'check_remote_block_number', calldata).unwrap();
+    assert(*result.at(0_u32) == 100, *result.at(0_u32));
+}
+
+#[test]
+fn test_start_roll_with_invoke() {
+    let deployed_contract_address = deploy_contract('storing_block_number', ArrayTrait::new()).unwrap();
+    assert(deployed_contract_address != 0, 'deployed_contract_address != 0');
+
+    let result = call(deployed_contract_address, 'retrieve_stored', ArrayTrait::new()).unwrap();
+    assert(*result.at(0_u32) == 0, *result.at(0_u32));
+    start_roll(100, deployed_contract_address);
+
+    let result = invoke(deployed_contract_address, 'store_block_number', ArrayTrait::new()).unwrap();
+
+    let result = call(deployed_contract_address, 'retrieve_stored', ArrayTrait::new()).unwrap();
+    assert(*result.at(0_u32) == 100, *result.at(0_u32));
+}
+
+#[test]
+fn test_start_roll_constructor() {
+    let class_hash = declare('storing_constructor_blk_number').unwrap();
+    let prepared = prepare(class_hash, ArrayTrait::new()).unwrap();
+
+    start_roll(100, prepared.contract_address);
+
+    let deployed_contract_address = deploy(prepared).unwrap();
+    let result = call(deployed_contract_address, 'retrieve_stored', ArrayTrait::new()).unwrap();
+    assert(*result.at(0_u32) == 100, *result.at(0_u32));
+}
+
+#[test]
+fn test_start_stop_roll() {
+    let deployed_contract_address = deploy_contract('simple', ArrayTrait::new()).unwrap();
+    assert(deployed_contract_address != 0, 'deployed_contract_address != 0');
+
+    let result = call(deployed_contract_address, 'check_block_number', ArrayTrait::new()).unwrap();
+    assert(*result.at(0_u32) == -1, *result.at(0_u32));
```

Comment:
Why the default value for block number is `-1` and not `0` for example?

Reply:
Not dictated by me, the starknet instance defaults dictate this one

Reply:
I think perhaps the first block would be 0

Reply:
I think that's in fact the case:
https://starkscan.co/block/0

Reply:
Okay, was mostly curious about the choice. We can keep `-1`.